### DISCUSSION TITLE
Convert `wav` to `aac` or `mp3` without saving it to the disk

### DISCUSSION THREAD

**Discussion by alirezanet** [naudio/NAudio#1193] [Q&A]
In my application, I have a small mixer that creates a wave file, I would like to convert it to mp3/acc before returning it to the user, and since it is hosted in Linux I probably can't use the `MediaFoundationEncoder`. 

```csharp
          var fileReaders = filePaths.Select(path => new AudioFileReader(path)).ToArray();
           var playlist = new ConcatenatingSampleProvider(fileReaders);

            _memoryStream = new MemoryStream();
            var writer = new WaveFileWriter(_memoryStream, playlist.WaveFormat);
            var buffer = new float[playlist.WaveFormat.SampleRate * playlist.WaveFormat.Channels];
            int samplesRead;
            while ((samplesRead = playlist.Read(buffer, 0, buffer.Length)) > 0)
            {
                writer.WriteSamples(buffer, 0, samplesRead);
            }
            writer.Flush();
            _memoryStream.Position = 0;
         
            // TODO: we need to return mp3 or ACC file
            return _memoryStream;
```

this is a simplified version of what I have right now ... does NAudio support in-memory conversion? I couldn't find any working example or documentation

**Comment by mdcdi1315** [comment]
@alirezanet There is preliminary support in NAudio V3 for the SoundFile library that supports reading/writing some common audio file formats such as OGG/FLAC/MP3. That should work for your Linux environment.

**Comment by markheath** [comment]
Docs for the cross-platform NAudio.SoundFile library, including with streams are here: https://github.com/naudio/NAudio/blob/main/Docs/CrossPlatformAudioFilesWithSoundFile.md
